chore(form):added form validation (#99)
telerik/blazor-ui: form/remote-validation/readme.md

# Remote (server, async) Validation upon Input in Form

> The samples in this folder showcase how you can use remote validation on the server to prevent inputs when they don't satisfy certain conditions that cannot be implemented through the standard `DataAnnotation` validation.

Reviewer: Prefer shorter and simpler sentences:

```suggestion
The samples in this folder showcase how to use remote validation on the server to prevent invalid user input. This is useful when validation conditions cannot be implemented through standard `DataAnnotation` validation.
```

## The General Concept

> In a real situation, the server validation may fail, the server may have a general error, be unavailable, or the client may be disconnected and HTTP requests to your WebAPI may fail. In such cases, you should notify the user that the operation failed and it needs to fix the input so that it passes.

Reviewer: Prefer lists for better readability:

```suggestion
In a real situation, server validation may fail for multiple reasons:
* invalid user input
* general server error
* server unavailable
* disconnected client
* failing HTTP requests to your WebAPI

In such cases, you should notify the user that the operation failed and suggest appropriate actions.
```

> Handling the errors as appropriate for your application.

Reviewer: Not sure what this sentence means and what is its value here.

## Sample Details

> In this sample, the WebAPI backend does not allow when a client chooses Defense classification to leave the Description field empty, in any other case the Description field is not required.

Reviewer: Split to shorter sentences.

```suggestion
In this sample, the WebAPI backend does not allow the user to leave the Description field empty if **Defense** classification is selected. In the other cases, the Description field is not required.
```

> The example shows how you can throw exceptions in the WebAPI backend with meaningful information, send those meaningful instructions/errors to the client where they can bubble up and be shown to the user.

Reviewer: ```suggestion
The example shows how you can throw exceptions in the WebAPI backend with meaningful information, send instructions/errors to the client where they can bubble up and be shown to the user.
```

The core concept is to return a status code higher than 400 to indicate an error, and to ensure the messages are meaningful to the human that will read them.

> You can read more about the general concepts of handling WebAPI errors in articles like the following ones:

Reviewer: ```suggestion
You can read more about the general concepts of handling WebAPI errors in articles such as:
```
